src/lib.rs

```diff
@@ -567,6 +581,61 @@ impl FrankaSim {
                 ..Default::default()
             },
         );
+        let mut robot_joint_indices: [Option<usize>; 7] = [None; 7];
+        let mut gripper_joint_indices = [None; 2];
+        let mut end_effector_link_index = None;
+        let num_joints = physics_client.borrow_mut().get_num_joints(panda_id);
+        for i in 0..num_joints {
+            let joint_info = physics_client.borrow_mut().get_joint_info(panda_id, i);
+            let joint_name = joint_info.joint_name;
+            if let Some(joint_index) = FrankaSim::ROBOT_JOINT_NAMES
+                .iter()
+                .position(|&x| x == joint_name)
+            {
+                robot_joint_indices[joint_index] = Some(i);
+            }
+            if let Some(joint_index) = FrankaSim::GRIPPER_JOINT_NAMES
+                .iter()
+                .position(|&x| x == joint_name)
+            {
+                gripper_joint_indices[joint_index] = Some(i);
+            }
+            if joint_info.link_name == FrankaSim::END_EFFECTOR_LINK_NAME {
+                end_effector_link_index = Some(i);
+            }
+        }
+        let end_effector_link_index = end_effector_link_index.unwrap_or_else(|| {
+            panic!(
+                "Could not find end-effector link: \"{}\" in URDF",
+                FrankaSim::END_EFFECTOR_LINK_NAME
+            )
+        });
+        fn turn_options_into_values<F: Fn(usize) -> String, const N: usize>(
+            array: [Option<usize>; N],
+            error_handler: F,
```

Comment:
This is not really a "handler" but more like a "message generator". How about allowing the callee to properly handle the error?

```rs
fn turn_options_into_value<const N: usize>(array: [Option<usize>; N]) -> Option<[usize;N]> { ... }

// or even

fn turn_options_into_value<const N: usize>(array: [Option<usize>; N]) -> Result<[usize;N], MyError> { ... }
```

Reply:
This is just a little helper function to avoid code duplication for handling robot and gripper joints. From my point of view there is no other option than to panic, as triggering this message means that your URDF does not contain the joints that you would expect.

Reply:
You could make `FrankaSim::new()` return `Option` or `Result`. `panic!` in libraries is a bit smelly. I'd say, let the user decide how they want to handle the error

Reply:
I agree with you. However, to do it properly you would need to introduce error handling not only in FrankaSim::new() but throughout the entire crate. As this would be quite some work and it is currently not needed, I would do this at another time.